ux/src/main/java/com/google/ar/sceneform/ux/BaseArFragment.java

```diff
@@ -628,27 +617,26 @@ public void onPeekTouch(HitTestResult hitTestResult, MotionEvent motionEvent) {
 
     @Override
     public void onUpdate(FrameTime frameTime) {
-        Frame frame = arSceneView.getArFrame();
-
-        AugmentedImageDatabase augmentedImageDatabase = getArSceneView().getSession().getConfig().getAugmentedImageDatabase();
-        boolean hasAugmentedImageDatabase = augmentedImageDatabase != null && augmentedImageDatabase.getNumImages() > 0;
-        if (hasAugmentedImageDatabase && onAugmentedImageUpdateListener != null) {
-            for (AugmentedImage augmentedImage : frame.getUpdatedTrackables(AugmentedImage.class)) {
-                onAugmentedImageUpdateListener.onAugmentedImageTrackingUpdate(augmentedImage);
-            }
-        }
+        if (getArSceneView() == null || getArSceneView().getSession() == null || getArSceneView().getArFrame() == null)
+            return;
 
         if (getInstructionsController() != null) {
-            boolean showPlaneInstructions = !arSceneView.hasTrackedPlane();
+            // Instructions for the Plane finding mode.
+            boolean showPlaneInstructions = !getArSceneView().hasTrackedPlane();
             if (getInstructionsController().isVisible(InstructionsController.TYPE_PLANE_DISCOVERY) != showPlaneInstructions) {
                 getInstructionsController().setVisible(InstructionsController.TYPE_PLANE_DISCOVERY, showPlaneInstructions);
             }
-            boolean showAugmentedImageInstructions = hasAugmentedImageDatabase
-                    && !arSceneView.isTrackingFullyAugmentImage();
+
+            // Instructions for the Augmented Image finding mode.
+            boolean showAugmentedImageInstructions = !getArSceneView().isTrackingAugmentedImage();
             if (getInstructionsController().isVisible(InstructionsController.TYPE_AUGMENTED_IMAGE_SCAN) != showAugmentedImageInstructions) {
                 getInstructionsController().setVisible(InstructionsController.TYPE_AUGMENTED_IMAGE_SCAN, showAugmentedImageInstructions);
             }
         }
+
+        for (AugmentedImage augmentedImage : getArSceneView().getUpdatedAugmentedImages()) {
+            onAugmentedImageUpdateListener.onAugmentedImageTrackingUpdate(augmentedImage);
```

Comment:
If I'm not overlooked something `onAugmentedImageUpdateListener` could be null. So we need a `null` check or an Optional.